certbot/tests/test_server.py

```diff
@@ -0,0 +1,209 @@
+import json
+import treq
+
+from twisted.internet.defer import fail, inlineCallbacks, succeed
+from twisted.protocols.loopback import _LoopbackAddress
+from twisted.trial.unittest import TestCase
+from twisted.web.client import ProxyAgent, URI
+from twisted.web.server import Site
+
+from txfake import FakeServer
+
+from uritools import uricompose
+
+from certbot.server import MarathonEventServer, Health
+
+
+class FakeServerAgent(ProxyAgent):
+    """
+    ProxyAgent uses the entire URI as the request path, which is the correct
+    thing to do when talking to a proxy but not for non-proxy servers.
+    """
+    def request(self, method, uri, headers=None, bodyProducer=None):
+        key = ('http-proxy', self._proxyEndpoint)
+        parsedURI = URI.fromBytes(uri)
+        return self._requestWithEndpoint(
+            key, self._proxyEndpoint, method, parsedURI, headers,
+            bodyProducer, parsedURI.originForm)
+
+
+class MarathonEventServerTest(TestCase):
```

Comment:
I prefer the form `TestMarathonEventServer` (with the `Test` at the front) for classes containing actual tests. (Some test runners (pytest included) treat any class that starts with `Test` as a test class, although pytest also collects any subclasses of `TestCase` it finds.)